crates/services/txpool_v2/src/pool.rs

```diff
@@ -0,0 +1,209 @@
+        SelectionAlgorithm,
+    },
+    storage::Storage,
+};
+
+/// The pool is the main component of the txpool service. It is responsible for storing transactions
+/// and allowing the selection of transactions for inclusion in a block.
+pub struct Pool<PSProvider, S: Storage, CM, SA> {
+    /// Configuration of the pool.
+    pub config: Config,
+    /// The storage of the pool.
+    storage: S,
+    /// The collision manager of the pool.
+    collision_manager: CM,
+    /// The selection algorithm of the pool.
+    selection_algorithm: SA,
+    /// The persistent storage of the pool.
+    persistent_storage_provider: PSProvider,
+    #[cfg(test)]
+    tx_id_to_storage_id: HashMap<TxId, S::StorageIndex>,
+}
+
+impl<PSProvider, S: Storage, CM, SA> Pool<PSProvider, S, CM, SA> {
+    /// Create a new pool.
+    pub fn new(
+        persistent_storage_provider: PSProvider,
+        storage: S,
+        collision_manager: CM,
+        selection_algorithm: SA,
+        config: Config,
+    ) -> Self {
+        Pool {
+            storage,
+            collision_manager,
+            selection_algorithm,
+            persistent_storage_provider,
+            config,
+            #[cfg(test)]
+            tx_id_to_storage_id: HashMap::new(),
+        }
+    }
+}
+
+impl<PS, View, S, CM, SA> Pool<PS, S, CM, SA>
+where
+    PS: AtomicView<LatestView = View>,
+    View: TxPoolPersistentStorage,
+    S: Storage,
+    CM: CollisionManager<Storage = S, StorageIndex = S::StorageIndex>,
+    SA: SelectionAlgorithm<Storage = S, StorageIndex = S::StorageIndex>,
+{
+    /// Insert transactions into the pool.
+    /// Returns a list of results for each transaction.
+    /// Each result is a list of transactions that were removed from the pool
+    /// because of the insertion of the new transaction.
+    #[instrument(skip(self))]
+    pub fn insert(
+        &mut self,
+        transactions: Vec<PoolTransaction>,
+    ) -> Result<Vec<Result<Vec<PoolTransaction>, Error>>, Error> {
+        let latest_view = self
+            .persistent_storage_provider
+            .latest_view()
+            .map_err(|e| Error::Database(format!("{:?}", e)))?;
+        Ok(transactions
+            .into_iter()
+            .map(|tx| {
+                #[cfg(test)]
+                let tx_id = tx.id();
+                self.check_pool_is_not_full()?;
+                self.config.black_list.check_blacklisting(&tx)?;
+                Self::check_blob_does_not_exist(&tx, &latest_view)?;
+                self.storage.validate_inputs(
+                    &tx,
+                    &latest_view,
+                    self.config.utxo_validation,
+                )?;
+                let collisions = self
+                    .collision_manager
+                    .collect_colliding_transactions(&tx, &self.storage)?;
+                let dependencies = self.storage.collect_transaction_dependencies(&tx)?;
+                let has_dependencies = !dependencies.is_empty();
```

Comment:
In this PR yes it could be possible but given the work on https://github.com/FuelLabs/fuel-core/pull/2193 it's needed. I think this PR is ok to have some unused things resolved by the next PRs.